rust_src/src/windows.rs

```diff
@@ -1990,4 +1992,36 @@ pub fn window_pixel_height_before_size_change(window: LispWindowValidOrSelected)
     window.pixel_height_before_size_change
 }
 
+/// Return a window currently displaying BUFFER-OR-NAME, or nil if none.
+/// BUFFER-OR-NAME may be a buffer or a buffer name and defaults to
+/// the current buffer.
+///
+/// The optional argument ALL-FRAMES specifies the frames to consider:
+///
+/// - t means consider all windows on all existing frames.
+///
+/// - `visible' means consider all windows on all visible frames.
+///
+/// - 0 (the number zero) means consider all windows on all visible
+///     and iconified frames.
+///
+/// - A frame means consider all windows on that frame only.
+///
+/// Any other value of ALL-FRAMES means consider all windows on the
+/// selected frame and no others.
+#[lisp_fn(min = "0")]
+pub fn get_buffer_window(buffer_or_name: LispObject, all_frames: LispObject) -> LispObject {
+    let buffer = if buffer_or_name.is_nil() {
+        Some(ThreadState::current_buffer_unchecked())
+    } else {
+        Fget_buffer(buffer_or_name).as_buffer()
+    };
+
+    if let Some(b) = buffer {
```

Comment:
It's preferred you do a match instead if you also have an `else` block on the `if let`